packages/atomic/src/components/common/refine-modal/refine-modal-common.tsx

```diff
@@ -120,13 +120,17 @@ export function getClonedFacetElements(
   );
 
   const allFacetsInOrderInDOM = root.querySelectorAll(allFacetTags.join(','));
-
-  allFacetsInOrderInDOM.forEach((facetElement, index) => {
+  let index = 0;
+  allFacetsInOrderInDOM.forEach((facetElement) => {
     const clone = facetElement.cloneNode(true) as BaseFacetElement;
-    clone.isCollapsed = facetShouldBeInitiallyCollapsed(
-      index,
-      collapseFacetsAfter
-    );
+    if (!facetElement.classList.contains('atomic-hidden')) {
```

Comment:
I think we should avoid checking whether a facet has `atomic-hidden` and/or checking whether a facet has `display: none`. Instead, I think we should expose something more explicit, like a method on the `atomic-facet` element (e.g., `isVisible()`).

Reply:
For examples on how to do methods on custom elements, see `setRenderFunction` on `atomic-result-list`.